### PR TITLE
Initial upload for event-detectors scripts

### PR DESCRIPTION
**Pull request by aasbra**
This script is intended for use by customers and the Operations Team, and includes validation and logging to help identify issues that need to be addressed.

This script only supports the following event detector types:
 *      LOW_LIMIT
 *      HIGH_LIMIT
 *      MULTISTATE_STATE
 *      BINARY_STATE
 *      NO_UPDATE
 *      UPDATE
If additional types are needed, this script can be updated.

### CODE REVIEW

**Comment by aasbra** [comment]
I'm converting this to a draft PR to give me a chance to make it compatible with Mango 5.1. I'll test and then merge those changes here.

**Comment by aasbra** [comment]
I added the feature detection Jared suggested and have tested this on 5.0.4 and 5.1.1.

File: create-event-detectors.js [hunk 1]
```diff
@@ -0,0 +1,549 @@
...
+        case "UPDATE":
+            verbose(`${type}: supported`);
+            return;
+        default:
+            if (type) throw new Error(`detectorType ${type}: Not Supported!`);
+            else throw new Error(`: No supported detectorType provided!`);
+    }
+}
+
+const verbose = (logMessage) => {
```
Comment (jazdw) [inline create-event-detectors.js]: I'm surprised calling this lambda works above this line. Functions are hoisted to the top of the file, a `const` is not.

**Review by jazdw — APPROVED** [review]
Thanks for sharing this, good documentation too.

File: create-event-detectors.js (hunk 1 above)
Comment (jazdw) [reply create-event-detectors.js]: Actually I guess this makes sense, it is in scope when it is called.

Comment (aasbra) [reply create-event-detectors.js]: That was a carry-over from the original script. I thought it was a bit odd, but I left it in. I'll re-write this as a simple function to make it more readable.

File: create-event-detectors.js [hunk 2]
```diff
@@ -0,0 +1,549 @@
...
+
+    /*
+    * VALIDATION PATTERN
+    * Perform the validation but do not set the value in the detector.
+    * The value will be set in the detector after it is created.
+    */
+
+    //Validations
+    /*
+    //Why are we checking every value and failing if any of them are missing?
```
Comment (aasbra) [inline create-event-detectors.js]: This comment block should have been removed. There were some validations that were incorrect from the original script that were wrong. I will remove this block.

**Review by jazdw — APPROVED** [review]

**Comment by jazdw** [comment]
@aasbra go ahead and merge it, let me know if you can't. I didn't realize my reviews were being dismissed when you pushed.

**Comment by aasbra** [comment]
@jazdw
It’s probably something I am missing, but I don’t see a way to rename the existing “create_event_detectors” directory in GitHub.  I was thinking of renaming it to something like “event-detector-management” or something similar. I've uploaded the latest versions of the create, edit, and delete scripts into that directory, but I think the directory should be renamed.

_edit-event-detectors.js_
 The edit script has not been reviewed yet.

_create-event-detectors.js_
This was modified slightly, adding a check for setResetDuration for Mango 5.1

_delete-event-detectors.js_
This was modified slightly with improvements from your feedback on the other scripts. Do we need to make any accommodations for different versions of Mango to restart data points after deleting the detector, or is that not necessary?

**Comment by jazdw** [comment]
> @jazdw It’s probably something I am missing, but I don’t see a way to rename the existing “create_event_detectors” directory in GitHub. I was thinking of renaming it to something like “event-detector-management” or something similar.

Git has no concept of tracking changes to directories, only files. Basically you just need to move all of the files in that directory to a new directory.

> _delete-event-detectors.js_ This was modified slightly with improvements from your feedback on the other scripts. Do we need to make any accommodations for different versions of Mango to restart data points after deleting the detector, or is that not necessary?

Yes I think that you probably need to reload the points if you delete the detectors in old versions of Mango.

**Comment by aasbra** [comment]
I updated this PR to:

1. Remove the poorly named folder create_event_detectors
2. Add a new folder called called event_detector_scripts
3. Move new scripts into the new folder

**Pull request merged**